internal/docker/builder.go

```diff
@@ -179,12 +178,23 @@ func (d *Builder) ConstructBlueprintsIfNotExist(bs []b.Blueprint) error {
 }
 
 func (d *Builder) ConstructBlueprints(bs []b.Blueprint) error {
-	var bpWg sync.WaitGroup
-	bpWg.Add(len(bs))
+	errc := make(chan []error, len(bs))
 	for _, bprint := range bs {
-		go d.construct(bprint, &bpWg)
+		go (func(bprint b.Blueprint) {
+			errc <- d.construct(bprint)
+		})(bprint)
 	}
-	bpWg.Wait()
+	var errs []error
+	for i := 0; i < len(bs); i++ {
+		errs = append(errs, <-errc...)
+	}
+	if len(errs) > 0 {
```

Comment:
Is this really the right check? Won't `errs` just be `[nil, nil, nil, ...]` in the happy path?

Reply:
I am appending `<-errc...` to `errs`, so if the returned error array (from the channel) is empty, nothing will be appended and the length of `errs` will not change.

Reply:
It's not super clear this is the case. Please add a comment to the append.

Reply:
I added a comment, is it explicit enough?